### DISCUSSION TITLE
Error Building stm-gap8 cpx example

### DISCUSSION THREAD

**Discussion by Hamedamz** [bitcraze/discussions#1517] [Q&A] [answered]
Hello,

I have an issue when I try to build the stm-gap8 cpx using the following command:

```
~/aideck-gap8-examples$ docker run --rm -v ${PWD}:/module bitcraze/aideck tools/build/make-example examples/other/stm_gap8_cpx all
Collecting numpy==1.22.3
  Using cached numpy-1.22.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (16.8 MB)
Installing collected packages: numpy
  Attempting uninstall: numpy
ERROR: imageio 2.17.0 has requirement pillow>=8.3.2, but you'll have pillow 7.2.0 which is incompatible.
    Found existing installation: numpy 1.24.3
    Uninstalling numpy-1.24.3:
      Successfully uninstalled numpy-1.24.3
Successfully installed numpy-1.22.3
gapy --target=ai_deck --platform=board --work-dir=/module/examples/other/stm_gap8_cpx/BUILD/GAP8_V2/GCC_RISCV_FREERTOS    run --image --binary=/module/examples/other/stm_gap8_cpx/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/stm_gap8_cpx  
gapy --target=ai_deck --platform=board --work-dir=/module/examples/other/stm_gap8_cpx/BUILD/GAP8_V2/GCC_RISCV_FREERTOS    run --flash --binary=/module/examples/other/stm_gap8_cpx/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/stm_gap8_cpx  
TARGET create
Open On-Chip Debugger 0.10.0+dev-gb84d97ec4 (2022-04-20-14:26)
Licensed under GNU GPL v2
For bug reports, read
	http://openocd.org/doc/doxygen/bugs.html
debug_level: 0

DEPRECATED! use 'adapter speed' not 'adapter_khz'
DEPRECATED! use 'adapter driver' not 'interface'
GAP8 INIT TARGET
Error: libusb_open() failed with LIBUSB_ERROR_NO_DEVICE
Error: no device found
Error: unable to open ftdi device with vid 15ba, pid 002a, description 'Olimex OpenOCD JTAG ARM-USB-TINY-H', serial '*' at bus location '*'

Flashing image with command:
gap8-openocd -d0 -c "gdb_port disabled; telnet_port disabled; tcl_port disabled" -c "script interface/ftdi/olimex-arm-usb-tiny-h.cfg; script target/gap8revb.tcl; script tcl/flash_image.tcl; script tcl/jtag_boot.tcl; gap_flash_raw_hyper /module/examples/other/stm_gap8_cpx/BUILD/GAP8_V2/GCC_RISCV_FREERTOS/target.board.devices.flash.img 50208 /gap_sdk/utils/openocd_tools; exit;"
Traceback (most recent call last):
  File "/gap_sdk/utils/gapy/gapy", line 162, in <module>
    main()
  File "/gap_sdk/utils/gapy/gapy", line 152, in main
    operationFunc(args, config, system)
  File "/gap_sdk/utils/gapy/run.py", line 189, in operationFunc
    raise RuntimeError('Runner has failed with value: %d' % status)
RuntimeError: Runner has failed with value: -1
make: *** [/gap_sdk/rtos/freeRTOS/vendors/gwt/rules/freeRTOS_rules.mk:442: flash_fs] Error 1
```

I'd appreciate any help.

**Comment by knmcguire** [comment]
I've just reopened the discussion. Could you post the solution instead so that others who had this same issue can benefit from that?

**Reply by Hamedamz** [reply]
It worked once after I updated Crazyflie and AIdeck firmware. Although I had updated the firmware and tested the wifi streaming once before in a different day and it was working, I used the cfclient to update the firmware again and then I flashed the wifi streaming example on the Aideck using `cfloader flash aideck_gap8_wifi_img_streamer_with_ap.bin deck-bcAI:gap8-fw -w [CRAZYFLIE_URI]`. Then I build stm-gap8 cpx example by running `docker run --rm -v ${PWD}:/module bitcraze/aideck tools/build/make-example examples/other/stm_gap8_cpx all` command and surprisingly it worked.

Now that I tried again to build the stm-gap8 cpx example it is not working and encounters the same error.

I am not sure but I assume updating the firmware or flashing the Aideck has a side effect that enables building the example.

**Reply by knmcguire** [reply]
This error is the key probably as it can't find the programmer: 
```
Error: unable to open ftdi device with vid 15ba, pid 002a, description 'Olimex OpenOCD JTAG ARM-USB-TINY-H', serial '*' at bus location '*'
```

Make sure all cables are connected, and double check the 10 pin lint cable that is connected to your aideck. Maybe there is a breakage in one of the wires in there

**Comment by zyxzxy** [comment]
I hope this message finds you well. I came across your discussion on GitHub regarding the JTAG scan chain issue with the Olimex ARM-USB-OCD-H adapter while working on the GAP8 platform ([link to the discussion](https://github.com/orgs/bitcraze/discussions/1517)). I am currently facing a very similar problem where OpenOCD reports a "Error: unable to open ftdi device with vid 15ba, pid 002a, description 'Olimex OpenOCD JTAG ARM-USB-TINY-H', serial '*' at bus location '*'" error, and I am struggling to resolve it.

I have tried several approaches, including adjusting the adapter speed, checking power and connections, and updating the OpenOCD configuration files, but the issue persists. Since you have encountered and perhaps resolved a similar problem, I would be incredibly grateful if you could share any insights or solutions that helped you overcome it.

Thank you very much for your time and assistance. Your guidance would be invaluable in helping me move forward with my project.

**Reply by gemenerik** [reply]
Thank you for reaching out! It sounds like the issue may be due to the `GAPY_OPENOCD_CABLE` environment variable, which defines the debugger OpenOCD will use. This variable is set when sourcing `ai_deck.sh` in the `gap_sdk/config` directory, and there’s also an export in `aideck-gap8-examples/tools/build/make-example` that may influence which debugger is used, depending on what you are running.

To resolve this, I recommend ensuring that `GAPY_OPENOCD_CABLE` is pointing to the debugger you’re using (`Olimex ARM-USB-OCD-H`) rather than `ARM-USB-TINY-H`. Adjusting this setting in both `ai_deck.sh` and in your workflow should help align OpenOCD with the correct debugger.

**Reply by zyxzxy** [reply, marked as answer]
Thank you for your reply. This problem was solved after I replaced a data cable. Now I am trying to reproduce the project of nanoflownet. Now I have encountered some new problems, can you help me to look at it? https://github.com/orgs/bitcraze/discussions/1633, thank you again for your reply!

**Reply by Bananazooka** [reply]
Hi! I've marked this issue has answered; thanks for opening a new discussion with your other issue - we'll discuss there.
